api/v1beta1/cryostat_conversion.go

```diff
@@ -0,0 +1,590 @@
+			CoreResources:       srcOpts.CoreResources,
+			DataSourceResources: srcOpts.DataSourceResources,
+			GrafanaResources:    srcOpts.GrafanaResources,
+		}
+	}
+	return dstOpts
+}
+
+func convertAuthPropertiesTo(srcProps *AuthorizationProperties) *operatorv1beta2.AuthorizationProperties {
+	var dstProps *operatorv1beta2.AuthorizationProperties
+	if srcProps != nil {
+		dstProps = &operatorv1beta2.AuthorizationProperties{
+			ClusterRoleName: srcProps.ClusterRoleName,
+			ConfigMapName:   srcProps.ConfigMapName,
+			Filename:        srcProps.Filename,
+		}
+	}
+	return dstProps
+}
+
+func convertSecurityOptionsTo(srcOpts *SecurityOptions) *operatorv1beta2.SecurityOptions {
+	var dstOpts *operatorv1beta2.SecurityOptions
+	if srcOpts != nil {
+		dstOpts = &operatorv1beta2.SecurityOptions{
+			PodSecurityContext:        srcOpts.PodSecurityContext,
+			CoreSecurityContext:       srcOpts.CoreSecurityContext,
+			DataSourceSecurityContext: srcOpts.DataSourceSecurityContext,
+			GrafanaSecurityContext:    srcOpts.GrafanaSecurityContext,
+		}
+	}
+	return dstOpts
+}
+
+func convertTargetDiscoveryTo(srcOpts *TargetDiscoveryOptions) *operatorv1beta2.TargetDiscoveryOptions {
+	var dstOpts *operatorv1beta2.TargetDiscoveryOptions
+	if srcOpts != nil {
+		dstOpts = &operatorv1beta2.TargetDiscoveryOptions{
+			BuiltInDiscoveryDisabled: srcOpts.BuiltInDiscoveryDisabled,
+		}
+	}
+	return dstOpts
+}
+
+func convertDatabaseOptionsTo(srcOpts *JmxCredentialsDatabaseOptions) *operatorv1beta2.JmxCredentialsDatabaseOptions {
+	var dstOpts *operatorv1beta2.JmxCredentialsDatabaseOptions
+	if srcOpts != nil {
+		dstOpts = &operatorv1beta2.JmxCredentialsDatabaseOptions{
+			DatabaseSecretName: srcOpts.DatabaseSecretName,
+		}
+	}
+	return dstOpts
+}
+
+func convertOperandMetadataTo(srcOpts *OperandMetadata) *operatorv1beta2.OperandMetadata {
+	var dstOpts *operatorv1beta2.OperandMetadata
+	if srcOpts != nil {
+		dstOpts = &operatorv1beta2.OperandMetadata{
+			DeploymentMetadata: convertResourceMetadataTo(srcOpts.DeploymentMetadata),
+			PodMetadata:        convertResourceMetadataTo(srcOpts.PodMetadata),
+		}
+	}
+	return dstOpts
+}
+
+func convertResourceMetadataTo(srcMeta *ResourceMetadata) *operatorv1beta2.ResourceMetadata {
+	var dstMeta *operatorv1beta2.ResourceMetadata
+	if srcMeta != nil {
+		dstMeta = &operatorv1beta2.ResourceMetadata{
+			Annotations: srcMeta.Annotations,
+			Labels:      srcMeta.Labels,
+		}
+	}
+	return dstMeta
+}
+
+// ConvertFrom converts from the Hub version (v1beta2) to this version.
+func (dst *Cryostat) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*operatorv1beta2.Cryostat)
+
+	// Copy ObjectMeta as-is
+	dst.ObjectMeta = src.ObjectMeta
+
+	// Convert existing Spec fields
+	convertSpecFrom(&src.Spec, &dst.Spec)
+
+	// Convert existing Status fields
+	convertStatusFrom(&src.Status, &dst.Status)
+
+	return nil
+}
+
+func convertSpecFrom(src *operatorv1beta2.CryostatSpec, dst *CryostatSpec) {
+	dst.Minimal = src.Minimal
```

Comment:
I'm in favour of removing the `minimal` config option: https://github.com/cryostatio/cryostat-helm/pull/122